File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
       int profit=0;
       int cost= prices[0];

       for(int i=1;i<prices.length;i++){
            profit= Math.max(profit,prices[i]-cost);
            cost= Math.min(cost,prices[i]);
       }
       return profit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = INT_MIN ,buy = prices[0]; // will buy first day stock 

        for(int i=0;i<prices.size();i++){
            int netProfit = prices[i]-buy; // buy it and sell it to track the profit
            profit = max(profit,netProfit); // update the max Profit
            buy = min(buy,prices[i]); // purchase stock at lower price to increase profit
        }


        return profit;
    }
};